### PR TITLE
Update `Nginx Ingress` entry to be more accurately `Ingress-Nginx`

### PR DESCRIPTION
**Pull request by joshpaulchan**
## What

- [x] updates the `Nginx Ingress Controller` entry to be `Ingress-Nginx`

## Why

I figured this was more accurate given the current copy on the surface refers to a different ingress controller, and an easy thing to fix!

### CODE REVIEW

**Comment by brandond** [comment]
This was consistent with upstream at the time we added it, but does appear that the upstream project has made an attempt to rebrand or at least be more consistent about branding. Note the changes in https://github.com/kubernetes/ingress-nginx/commit/1614027cd42aa3fac53d62afc235b86adec7543d

**Comment by joshpaulchan** [comment]
@brandond Sweet! Glad it's further differentiating itself from the proprietary nginx ingress controller. Per this PR, let me know if I should tweak it in any way!

File: docs/architecture.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ RKE2 brings together a number of Open Source technologies to make this all work:
 - [containerd][io-containerd]/[cri][gh-cri-api]
 - [CNI][gh-cni]: Canal ([Calico][org-projectcalico] &amp; [Flannel][gh-flannel]), [Cilium][io-cilium] or [Calico][org-projectcalico]
 - [CoreDNS][io-coredns]
-- [NGINX Ingress Controller][io-ingress-nginx]
+- [Ingress-Nginx Controller][io-ingress-nginx]
```
Comment (brandond) [inline docs/architecture.md:32]: It looks like upstream is not hyphenating
```suggestion
- [Ingress Nginx Controller][io-ingress-nginx]
```

**Review by brandond — CHANGES_REQUESTED** [review]

File: docs/architecture.md (hunk 1 above)
Comment (brandond) [reply docs/architecture.md:32]: thanks for suggesting this change!

**Comment by brandond** [comment]
thanks for pointing this out!

**Review by brandond — APPROVED** [review]

**Review by dereknola — APPROVED** [review]

**Review by manuelbuil — APPROVED** [review]

**Pull request merged**